cwms-data-api/src/main/java/cwms/cda/api/LevelsController.java

```diff
@@ -581,20 +648,76 @@ private LocationLevel updatedClearedFields(String body, String format,
         JavaType javaType = om.getTypeFactory().constructType(LocationLevel.class);
         BeanDescription beanDescription = om.getSerializationConfig().introspect(javaType);
         List<BeanPropertyDefinition> properties = beanDescription.findProperties();
-        LocationLevel retVal = new LocationLevel.Builder(existingLevel).build();
-        try {
-            for (BeanPropertyDefinition propertyDefinition : properties) {
-                String propertyName = propertyDefinition.getName();
-                JsonNode propertyValue = root.findValue(propertyName);
-                if (propertyValue != null && "".equals(propertyValue.textValue())) {
-                    retVal = new LocationLevel.Builder(retVal)
-                                    .withProperty(propertyName, null).build();
+        if (existingLevel instanceof ConstantLocationLevel) {
+            ConstantLocationLevel constantLevel = (ConstantLocationLevel) existingLevel;
+            ConstantLocationLevel retVal = new ConstantLocationLevel.Builder(constantLevel).build();
+            try {
+                for (BeanPropertyDefinition propertyDefinition : properties) {
+                    String propertyName = propertyDefinition.getName();
+                    JsonNode propertyValue = root.findValue(propertyName);
+                    if (propertyValue != null && "".equals(propertyValue.textValue())) {
+                        retVal = new ConstantLocationLevel.Builder(retVal)
+                                .withProperty(propertyName, null).build();
+                    }
                 }
+            } catch (NullPointerException e) {
+                //gets thrown if required field is null
+                throw new IllegalArgumentException(e.getMessage());
```

Comment:
we should include the exception so that the stacktrace can hit the log and people can trace issues, should they happen.

also what is throwing null that we can't check for directly?